File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        

        def dfs(i,t):
            if i==len(nums):return t
            return dfs(i+1,t) + dfs(i+1,t^nums[i])
        return dfs(0,0)
            

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.java
class Solution {
    public int subsetXORSum(int[] nums) {
         return dfs(nums,0,0);  
    }

    public int dfs(int[] nums,int i,int t){
        if(i==nums.length) return t;
        return dfs(nums,i+1,t) + dfs(nums,i+1,t^nums[i]);
    }
}